modules/cnf-numa-resource-scheduling-strategies.adoc

```diff
@@ -0,0 +1,70 @@
+// Module included in the following assemblies:
+//
+// *scalability_and_performance/cnf-numa-aware-scheduling.adoc
+
+:_module-type: CONCEPT
+[id="cnf-numa-resource-scheduling-strategies_{context}"]
+= NUMA Resource scheduling strategies 
+
+When scheduling high-performance workloads, the secondary scheduler can employ different strategies to determine which NUMA node within a chosen worker node will handle the workload. The supported strategies in {product-title} include `LeastAllocated`, `MostAllocated`, and `BalancedAllocation`. Understanding these strategies helps optimize workload placement for performance and resource utilization.
+
+When a high-performance workload is scheduled in a NUMA-aware cluster, the following steps occur: 
+
+. The scheduler first selects a suitable worker node based on cluster-wide criteria for example taints, labels, or resource availability.
+
+. After a worker node is selected, the scheduler evaluates its NUMA nodes and applies a scoring strategy to decide which NUMA node will handle the workload.
+
+. Once a workload is scheduled, the selected NUMA node’s resources are updated to reflect the allocation.
+
+The default strategy applied is the `LeastAllocated` strategy. This assigns workloads to the NUMA node with the most available resources that is the least utilized NUMA node. The goal of this strategy is to spread workloads across NUMA nodes to reduce contention and avoid hotspots.
+
+.Example
+Assume a worker node has two NUMA nodes, and the workload requires 4 vCPUs and 8 GB of memory:
+
+[cols="5,2,2,2,2,2", options="header"]
+|===
+|NUMA Node |Total CPUs |Used CPUs |Total Memory (GB) |Used Memory (GB) |Available Resources
+|NUMA 1 |16 |12 |64 |56 |4 CPUs, 8 GB Memory
+|NUMA 2 |16 |6 |64 |24 |10 CPUs, 40 GB Memory
+|===
+
+NUMA 2 has more available resources (10 CPUs, 40 GB memory) compared to NUMA 1 (4 CPUs, 8 GB memory) and therefore with the default strategy in place the workload is assigned to NUMA 2.
+
+
+The `MostAllocated` strategy assigns workloads to the NUMA node with the least available resources that is the the most utilized NUMA node. The goal of this strategy is to consolidate workloads on fewer NUMA nodes, leaving other NUMA nodes idle or with free resources for energy efficiency or other workloads.
+
+.Example
+Using the same initial NUMA node states as earlier:
+
+[cols="2,2"]
+|===
+|NUMA Node |Available Resources
+|NUMA 1 |4 CPUs, 8 GB Memory
+|NUMA 2 |10 CPUs, 40 GB Memory
+|===
+
+NUMA 1 has fewer available resources compared to NUMA 2 and therefore with the `MostAllocated` strategy in place the workload is assigned to NUMA 1, further consolidating usage.
+
+The `BalancedAllocation` strategy assigns workloads to the NUMA node with the most balanced resource utilization across CPU and memory. The goal of this strategy is to avoid scenarios where one resource for example CPU is heavily utilized while another for example memory is underutilized.
```

Comment:
```suggestion
The `BalancedAllocation` strategy assigns workloads to the NUMA node with the most balanced resource utilization across CPU and memory. The goal of this strategy is to avoid scenarios where one resource, for example CPU, is heavily utilized while another for example memory is underutilized.
```